src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```diff
@@ -189,14 +200,21 @@ def prepare_or_run_pipeline(
                 deployment=deployment, step_name=pipeline_step_name
             )
 
-        # Build entrypoint command and args for the orchestrator pod.
-        # This will internally also build the command/args for all step pods.
-        command = SkypilotOrchestratorEntrypointConfiguration.get_entrypoint_command()
+        if self.config.configure_step_resources:
+            # Run each step in a separate VM if configured.
+            # Build entrypoint command and args for the orchestrator VM.
+            # This will internally also build the command/args for all step pods.
+            command = SkypilotOrchestratorEntrypointConfiguration.get_entrypoint_command()
+            args = SkypilotOrchestratorEntrypointConfiguration.get_entrypoint_arguments(
+                run_name=orchestrator_run_name,
+                deployment_id=deployment.id,
+            )
+        else:
+            command = PipelineEntrypointConfiguration.get_entrypoint_command()
```

Comment:
here we should raise warning if they've done step specific configurations...

actually maybe we can automatically make the switch if we detect step specific config?